sklearn/linear_model/tests/test_sag.py

```diff
@@ -449,7 +449,8 @@ def test_sag_regressor_computed_correctly(csr_container):
     n_features = 10
     n_samples = 40
     max_iter = 100
-    tol = 0.000001
+    max_iter_sparse = 1000
+    tol = 1.5e-4
```

Comment:
Why did you increase the tolerance? The test was too slow otherwise?

Assuming that the test can run in less than a second, I would rather use a very large `max_iter=10_000` and a small tolerance (e.g. `1e-7` or even lower) and make the assertions stricter.

Also I would replace `assert_array_almost_equal` / `assert_almost_equal` by `assert_all_close` from scikit-learn and maybe pass explicitly values for `rtol=1e-5` and `atol=1e-5` (to values set as small as possible).

Reply:
https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/linear_model/_ridge.py#L857
If we won't use bigger `tol` we will end up with solver other than sag.

Reply:
Also, we check if max_iter is None and only then choose sag solver, because `solver requires many iterations to fit`. Then max_iter will become 1000.
 https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/linear_model/_sag.py#L246.
Nevertheless, if user choose max_iter=1001, we will not use sag. But 1001 will not be worse in terms of `solver requires many iterations to fit`. Maybe it will be consistent with message content to change if statement here https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/linear_model/_ridge.py#L857 to
`elif self.solver == "sag" and (self.max_iter is None or self.max_iter >= 1000) and self.tol > 1e-4:`?